### DISCUSSION TITLE
.rst file result precision question

### DISCUSSION THREAD

**Discussion by xiaomaiAI** [pyansys/pymapdl-reader#54] [Q&A] [answered]
Hi,
    Is rst result datas store in float type? I took these datas compared with ANSYS solution, I found only 6 or 7 significant digits
are correct.  thanks.

**Comment by akaszynski** [comment, marked as answer]
In the header file describing Ansys resutls, `fdresu.inc`, there are detailed descriptions of each data type output from the result file.  It appears except for special cases, the result records are stored in double precision.

These header files are included in the unified install and can be found at:
```
.../ansys/customize/include/fdresu.inc
```
Path may vary on Windows.

---

Since I'm not using any of Ansys's post-processing routines in this module, it's possible that the difference you're seeing is due to variations in how we implemented certain routines.  For example, when computing the principal stress `ansys-mapdl-reader` uses `np.linalg._umath_linalg.eigvalsh_lo` for each 3x3 stress tensor, which may generate different results than Ansys's approach.  Same with euler angle rotations.

However for results that don't involve post-processing, the results should be `1:1` since `ansys-mapdl-reader` directly reads in the results from the result files.

**Comment by mikerife** [comment]
Hi @xiaomaiAI & @akaszynski Sorry about being so late to the conversation.  And I hope this actually helps.  But starting at version 14.5 MAPDL stores element results in single precision (stresses, strains, nodal forces, etc).  Nodal results (displacements etc) and reactions remain double precision.  
Mike
